File: Shared/ExceptionUtils.vb
'*********************************************************************************************
' ST-Sim: A SyncroSim Module for the ST-Sim State-and-Transition Model.
'
' Copyright © 2007-2017 Apex Resource Management Solution Ltd. (ApexRMS). All rights reserved.
'
'*********************************************************************************************
Imports System.Globalization

Module ExceptionUtils

    ''' <summary>
    ''' Throws an argument exception
    ''' </summary>
    ''' <param name="message"></param>
    Public Sub ThrowArgumentException(message As [String])
        ThrowArgumentException(message, New Object() {})
    End Sub

    ''' <summary>
    ''' Throws an argument exception
    ''' </summary>
    ''' <param name="message"></param>
    Public Sub ThrowArgumentException(message As [String], ParamArray args As Object())
        Throw New ArgumentException(String.Format(CultureInfo.InvariantCulture, message, args))
    End Sub

    ''' <summary>
    ''' Throws an invalid operation exception
    ''' </summary>
    ''' <param name="message"></param>
    Public Sub ThrowInvalidOperationException(message As [String], ParamArray args As Object())
        Throw New InvalidOperationException(String.Format(CultureInfo.InvariantCulture, message, args))
    End Sub

End Module

File: Shared/ExceptionUtils.cs
// ST-Sim: A SyncroSim Module for the ST-Sim State-and-Transition Model.
// Copyright © 2007-2018 Apex Resource Management Solution Ltd. (ApexRMS). All rights reserved.

using System;
using System.Globalization;

namespace SyncroSim.STSim
{
    internal static class ExceptionUtils
    {
        public static void ThrowArgumentException(string message)
        {
            ThrowArgumentException(message, new object[0]);
        }

        public static void ThrowArgumentException(string message, params object[] args)
        {
            throw new ArgumentException(string.Format(CultureInfo.InvariantCulture, message, args));
        }

        public static void ThrowInvalidOperationException(string message, params object[] args)
        {
            throw new InvalidOperationException(string.Format(CultureInfo.InvariantCulture, message, args));
        }
    }
}
